tests/admin_views/admin.py

```diff
@@ -150,6 +150,10 @@ def has_change_permission(self, request, obj=None):
         """ Only allow changing objects with even id number """
         return request.user.is_staff and (obj is not None) and (obj.id % 2 == 0)
 
+    def has_view_permission(self, request, obj=None):
+        """ Only allow viewing objects with even id number """
+        return request.user.is_staff and (obj is not None) and (obj.id % 2 == 0)
```

Comment:
I haven't run the tests, so I could be off here, but if having the _change_ permission effectively grants you the _view_ permission for backwards compatibility, is having the same criteria here a good idea? It could potentially mask problems with the new _view_ permission in the tests as the user would always have both.